src/groovy/org/pih/warehouse/api/StockMovement.groovy

```diff
@@ -482,4 +482,13 @@ class StockMovement {
         def g = ApplicationHolder.application.mainContext.getBean('org.codehaus.groovy.grails.plugins.web.taglib.ApplicationTagLib')
         return "${g.message(code: 'enum.' + status?.getClass()?.getSimpleName() + '.' + status)}"
     }
+
+    Boolean canUserRollbackApproval(User user) {
+        // Approval can be rolled back by user who is approver or requestor
+        return approvers?.contains(user) || requestedBy == user
+    }
+
+    Boolean isInApprovalState() {
+        return requisition?.status in [RequisitionStatus.APPROVED, RequisitionStatus.REJECTED]
```

Comment:
how about `PENDING_APPROVAL`?

Reply:
I guess never mind, after going through the code again, I probably understand now.

Reply:
We can rollback only in APPROVED or REJECTED status